src/home/homed-home.c

```diff
@@ -0,0 +1,2716 @@
+        case HOME_UNFIXATED:
+        case HOME_INACTIVE:
+        case HOME_ACTIVE:
+                break;
+        default:
+                return sd_bus_error_setf(error, BUS_ERROR_HOME_BUSY, "An operation on home %s is currently being executed.", h->user_name);
+        }
+
+        r = home_ratelimit(h, error);
+        if (r < 0)
+                return r;
+
+        return home_authenticate_internal(h, secret, state == HOME_ACTIVE ? HOME_AUTHENTICATING_WHILE_ACTIVE : HOME_AUTHENTICATING, error);
+}
+
+static int home_deactivate_internal(Home *h, bool force, sd_bus_error *error) {
+        int r;
+
+        assert(h);
+
+        r = home_start_work(h, force ? "deactivate-force" : "deactivate", h->record, NULL);
+        if (r < 0)
+                return r;
+
+        home_set_state(h, HOME_DEACTIVATING);
+        return 0;
+}
+
+int home_deactivate(Home *h, bool force, sd_bus_error *error) {
+        assert(h);
+
+        switch (home_get_state(h)) {
+        case HOME_UNFIXATED:
+        case HOME_ABSENT:
+        case HOME_INACTIVE:
+                return sd_bus_error_setf(error, BUS_ERROR_HOME_NOT_ACTIVE, "Home %s not active.", h->user_name);
+        case HOME_LOCKED:
+                return sd_bus_error_setf(error, BUS_ERROR_HOME_LOCKED, "Home %s is currently locked.", h->user_name);
+        case HOME_ACTIVE:
+                break;
+        default:
+                return sd_bus_error_setf(error, BUS_ERROR_HOME_BUSY, "An operation on home %s is currently being executed.", h->user_name);
+        }
+
+        return home_deactivate_internal(h, force, error);
+}
+
+int home_create(Home *h, UserRecord *secret, sd_bus_error *error) {
+        int r;
+
+        assert(h);
+
+        switch (home_get_state(h)) {
+        case HOME_INACTIVE:
+                if (h->record->storage < 0)
+                        break; /* if no storage is defined we don't know what precisely to look for, hence
+                                * HOME_INACTIVE is OK in that case too. */
+
+                if (IN_SET(user_record_test_image_path(h->record), USER_TEST_MAYBE, USER_TEST_UNDEFINED))
+                        break; /* And if the image path test isn't conclusive, let's also go on */
+
+                _fallthrough_;
+        case HOME_UNFIXATED:
+                return sd_bus_error_setf(error, BUS_ERROR_HOME_EXISTS, "Home of user %s already exists.", h->user_name);
+        case HOME_ABSENT:
+                break;
+        case HOME_ACTIVE:
+        case HOME_LOCKED:
+        default:
+                return sd_bus_error_setf(error, BUS_ERROR_HOME_BUSY, "Home %s is currently being used, or an operation on home %s is currently being executed.", h->user_name, h->user_name);
+        }
+
+        if (h->record->enforce_password_policy == false)
+                log_debug("Password quality check turned off for account, skipping.");
+        else {
+                r = quality_check_password(h->record, secret, error);
```

Comment:
This might seem like a good idea, but does not work in practice. We had this fight a few times in Fedora, where anaconda developers and some security folks wanted to enforce password "quality". (And I use quotes because the quality of the checks in pwquality is just bad.) Generally, the checks are mostly pointless and enforce a 90's mentality of hard-to-memorize and hard-to-use passwords, while rejecting many strong passwords that don't look randomly generated. Apart from making users frustrated, this makes QA and debugging harder, because one has to jump through hoops to set a "test" password like "123". Every time changes that try to enforce this were reverted.

What works is to provide a hint: you enter a password, and the machine tells you "weak/mediu/strong", or "this looks weak, do you want to add more symbols?". Currently anaconda requires "done" to be pressed twice if the password is weak. The crucial part is that it is a hint, not a barrier.

The checks needs to be moved to the client side and made into hints. If CLI is not interactive, a switch like '--force' or '--I-like-my-password-thankyouverymuch' might be necessary.

Reply:
yupp, this already exists, use `--enforce-password-policy=no` (or just `-P`) when creating/changing the record, and from that moment on the user doesn't get any such policy enforced anymore. I mean, that pw policy drove me nuts while hacking this, so I of course made sure it's trivially easy to turn it off. Moreover the client tool will actually help you with this, when the error comes back from homed, and tell you to run teh command again with `-P` and all will be good. 

So we are strict by default, but in a very friendly way, and excluding a user from this is sticky, you just need to turn this on once for it, and the user is excluded for good.

So yeah, I'd expect anaconda/gnome/... to just disable the enforcement too (it's just setting a boolean json field in the record to false) and do it better on the client side.